nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.rest.management.elasticsearch;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.nuxeo.ecm.core.api.security.SecurityConstants.SYSTEM_USERNAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.ACTION_NAME;
+import static org.nuxeo.elasticsearch.bulk.IndexAction.INDEX_UPDATE_ALIAS_PARAM;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.query.sql.NXQL;
+import org.nuxeo.ecm.core.repository.RepositoryService;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.elasticsearch.api.ElasticSearchAdmin;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Endpoint to administer Elasticsearch
+ *
+ * @since 11.1
+ */
+@WebObject(type = "elasticsearch")
+public class ElasticsearchObject extends DefaultObject {
+
+    /**
+     * Performs an ES indexing on documents of the given repository and that match the NXQL query.
+     *
+     * @see #performIndexing(String, String)
+     */
+    @POST
+    @Path("{repositoryName}/reindex")
+    public Response doIndexing(@PathParam("repositoryName") String repositoryName, @QueryParam("query") String query) {
```

Comment:
What do you think to change `repositoryName` to a `QueryParam`? We have several places where we fallback on default repository if we don't explicitly provide a repository name. Note: This will be the case for at least 90% of environment.

In the same idea, we could do this change for other endpoints having such behavior (maybe with more code, don't know if ES services handles this default thing). But this will hide the _for all_ action 🤔 or we can handle such case with a wildcard, for instance `repositoryName = *`

Reply:
yep, for the first proposition, i through about it, and i told my self perhaps if it's an error of the user (missing the query param in his url) then we don't need to launch an indexing on the default repository.  i think the ES service hold this behaviour, let me check i don't remember if it's only on automation.

good idea 👍 for the wildcard indexing (the all exiting repositories ;)) i will make some tests and i will tell you

Reply:
if we don't provide a repository name, the default one will be taken ; [here](https://github.com/nuxeo/nuxeo/blob/564aa891032e778b757460f449792cf45cc6d862/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/message/BulkCommand.java#L175) and [here](https://github.com/nuxeo/nuxeo/blob/564aa891032e778b757460f449792cf45cc6d862/nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkServiceImpl.java#L98) i will modify the repositoryName to QueryParam instead of PathParam, but i don't know if we should add the Document and its children indexing, the idea is to transform the whole resources to QueryParam :
` @POST
    @Path("reindex")
    public Response doIndexing(@QueryParam("repositoryName") String repositoryName, @QueryParam("query") String query) {
        return performIndexing(repositoryName, query);
    }`

Reply:
Again, ask the people who will be using it.

If we decide to keep it, IMO, `documentId` should be a PathParam, which will solve your issue: 
```
@Path("reindex/{documentId}")
```

And you can still use an underlying NXQL query.

Reply:
yep i see, i was in favour to put all as queryParam and to correct the issue i was going to make   `@Path("reindex/{documentId}")` but it's a little disturbing, because i see the repo as the high level, it's means we access to the repo and then the document. 
but to simplify and to keep it, i will do it as that ;)

Reply:
Agree with @RSalem07 about the repository being a high level ressource, putting the `documentId` in the path whereas repository is not looks very weird to me. You'll end with:
```
curl -XPOST http://NUXEO/site/management/elasticsearch/reindex/00000000-0000-0000-0000-000000000000?repositoryName=default
```
Whereas NXQL can be independent from the repository on which we run it, the `documentId` is linked to repository and shouldn't be before/higher than the repository in the API.

Regarding the point above, only solution I see is:
```
@Path("reindex")
public Response reindex(@QueryParam("repositoryName") repositoryName, @QueryParam("query") query, @QueryParam("documentId") documentId) 
```
With this we need to handle the two cases depending on documentId presence.

Reply:
yes i was thinking about doing it like that, as you mention, if the document is present the query will be ignored, if we decide to put document as more relevant for us. it's why i didn't do it yesterday :)